tap_salesforce/salesforce/__init__.py

```diff
@@ -513,3 +523,100 @@ def get_blacklisted_fields(self):
             raise TapSalesforceException(
                 "api_type should be REST or BULK was: {}".format(
                     self.api_type))
+
+    def date_columns_in_filter(self, filter):
+        date_cols = set()
+        for f in filter["filters"]:
+            if f["rhs"]["litType"] == "date":
+                date_cols.add(f["lhs"]["name"])
+
+        return date_cols
+
+    def get_date_column_types(self, date_cols_in_filter):
+        response = self.describe()
+
+        date_col_types = dict()
+
+        if self.source_type == 'object':
+            fields = response['fields']
+            for f in fields:
+                if f['name'] in date_cols_in_filter:
+                    date_col_types[f['name']] = f['type']
+        elif self.source_type == 'report':
+            fields = response['reportExtendedMetadata']['detailColumnInfo']
+            for f in fields.values():
+                if f['label'] in date_cols_in_filter:
+                    date_col_types[f['label']] = f['dataType']
+
+        return date_col_types
+
+    def filter_sql(self, filter, date_col_types):
+        if (filter["filterType"] == "statement"):
+            return self.filter_statement_sql(filter, date_col_types)
+        else:
+            filters = filter["filters"]
+
+            group_filters_sql = []
+            for f in filters:
+                child_sql = self.filter_sql(f, date_col_types)
+                if child_sql is not None:
+                    group_filters_sql.append(child_sql)
+
+            if len(group_filters_sql) > 0:
+                group_op_sql = " {} ".format(filter["op"])
+                return "({})".format(group_op_sql.join(group_filters_sql))
+            else:
+                return None
+
+    def filter_statement_sql(self, statement, date_col_types):
+        lhs_sql = self.filter_operand_sql(statement["lhs"])
+        op_sql = self.filter_op_sql(statement["op"])
+
+        if "rhs" in statement:
+            rhs_sql = self.filter_operand_sql(statement["rhs"])
+
+            if statement["rhs"].get("litType") == 'date':
+                # Salesforce has date and datetime columns. Only datetime columns can include time, or query will throw an error
+                if date_col_types.get(lhs_sql) == 'datetime':
+                    # salesforce only store date in utc and needs Z at the end instead of +00:00, our filters need to match that
+                    rhs_sql = f"{datetime.fromisoformat(rhs_sql).replace(tzinfo=timezone.utc).isoformat().replace('+00:00', 'Z')}"
+                return f"({lhs_sql} {op_sql} {rhs_sql})"
+            if statement["rhs"].get("litType") == 'number' or statement["rhs"].get("litType") == 'boolean':
```

Comment:
@ChrisLing1 Also made a change here - numbers and boolean can't be wrapped in quotes